pkg/controller/flow/reconcile_test.go

```diff
@@ -255,3 +251,50 @@ func getOwnerReference() metav1.OwnerReference {
 		BlockOwnerDeletion: &falseVal,
 	}
 }
+
+// wrappedClient is a wrapper around a FakeClient. It is used to manipulate Create() and List()
+// responses.
+// TODO: Replace this with Reactors, once the FakeClient uses Reactors.
+type wrappedClient struct {
```

Comment:
@grantr I'm fairly sure this isn't what you meant by "use a mock", but it is how I interpreted it. What should this look like?

Reply:
The controller-runtime fake client uses the object tracker from client-go: https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/client/fake/client.go#L48, which should mean that for objects using `Name`, this should be necessary, based on my understanding here: https://github.com/kubernetes/client-go/blob/master/testing/fixture.go#L332.
It does not look like the tracker supports `GenerateName`, but I don't think that matters in this case. Did you try this without wrapped client?

Reply:
I initially created WrappedClient to handle the Channel's status. Initially, I couldn't test the changes to Feeds because the Channel needs to have an Internal Domain, https://github.com/knative/eventing/blob/ac3b4d5d3a7a702ede03c0f2f00ebb48463b18f4/pkg/controller/flow/reconcile.go#L128. 

Without the wrapped client, the Internal Domain would be the empty string, causing it to skip the reconcileFeed() method, which is where my changes went. The WrappedClient sets the Internal Domain on line 286.

Then I found that I couldn't list anything with the FakeClient or it would panic. I added the code to fake Feed listing (as that is all the test currently cares about). Looking in the [Feed Controller](https://github.com/knative/eventing/blob/ac3b4d5d3a7a702ede03c0f2f00ebb48463b18f4/pkg/controller/feed/reconcile.go#L520), I see that it was solved there by adding a useless in production parameter to the ListOptions, so I have switched to do that instead, to keep the style consistent across controllers.